From 46a622ae51d66b70370fc3b8cd5c96657708b7c6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Mon, 12 Feb 2018 16:25:41 +0100
Subject: [PATCH 4/4] cachunker: round the expected average chunk size (#135)

For some reason, in F28, we get 4095 as the result of the expression.
I would expect to always get 4096, but in principle 4096- 1/2 ulp is also a
correct answer, and then when truncating, we would get 4095. So let's round the
result to get the expected answer.

Fixes #133.

NOTE: Modified when cherry-picked to not used log_debug.
---
 src/cachunker.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/cachunker.c b/src/cachunker.c
index 138cfd4..d1f583e 100644
--- a/src/cachunker.c
+++ b/src/cachunker.c
@@ -19,7 +19,7 @@ int ca_chunker_set_size(
 
         if (avg_size == 0) {
                 if (min_size != 0 && max_size != 0)
-                        avg_size = (size_t) pow(2, (log2(min_size) + log2(max_size)) / 2);
+                        avg_size = (size_t) lrint(pow(2, (log2(min_size) + log2(max_size)) / 2));
                 else if (min_size != 0)
                         avg_size = min_size * 4;
                 else if (max_size != 0)
@@ -80,8 +80,8 @@ int ca_chunker_set_size(
 
         c->discriminator = discriminator;
 
-        /* fprintf(stderr, "Setting min/avg/max chunk size: %zu/%zu/%zu.\n", */
-        /*         c->chunk_size_min, c->chunk_size_avg, c->chunk_size_max); */
+        fprintf(stderr, "Setting min/avg/max chunk size: %zu / %zu / %zu\n",
+                c->chunk_size_min, c->chunk_size_avg, c->chunk_size_max);
 
         return 0;
 }
-- 
2.18.0

